Add Integration test in python example with docs (#881)
pulumi/examples: testing-integration-py/README.md

# Integration Testing of Pulumi programs in Python

This integration test is using Pulumi automation api [Automation API](https://www.pulumi.com/blog/automation-api/) and [Python Unittest](https://docs.python.org/3/library/unittest.html)  to simulate integration test in Pulumi without native python integration library. 

| UnitTest | Integration Test |Target|
|--|--|--|
| YES | YES|Input|
| No | YES|Output|

## Test Case
To create a s3 bucket and upload a file

 1. Create the stack with a bucket
 3. Verify S3 bucket name and region in Output
 4. Using client to check the S3 name again
 5. Using client to upload a file
 6. Deleting that file
 7. Delete the stack with the bucket

## Precondition

you meed create the credential file yourself. By default, its location is at ~/.aws/credentials, and And access key has the permission for S3 bucket.
```
[default]
aws_access_key_id = YOUR_ACCESS_KEY
aws_secret_access_key = YOUR_SECRET_KEY
```

##  Running the tests
```
pip install -r requirements.txt
python -m unittest test_s3_it.py
```

## Test Life Cycle

>  - create a stack for all the use cases, to get all the output from stack
>  - assert any output value you defined in advance.
>  - In the end, don't forget destroy the stack. 

Reviewer: ```suggestion
 - Create a stack and export the desired outputs.
 - Validate any output values you defined in advance.
 - In the end, don't forget to destroy the stack. 
```

```
from pulumi.x import automation as auto

class TestS3(unittest.TestCase):  
  
    @classmethod  
    def setUpClass(cls) -> None:  
      ....
        cls.stack = auto.create_or_select_stack(stack_name=cls.STACK_NAME, work_dir=cls.WORK_DIR) 

>         cls.outputs = cls.stack.outputs() 

Reviewer: Might be helpful to specify that an `up` is necessary before pulling outputs.
```suggestion
		cls.stack.up(output=print)
        cls.outputs = cls.stack.outputs() 
```

       ...
          
    @classmethod  
    def tearDownClass(cls) -> None:  

>         cls.stack.destroy(on_output=print)  

Reviewer: You might want to remove the stack completely at the end.

```suggestion
        cls.stack.destroy(on_output=print)
        cls.stack.workspace.remove_stack(cls.STACK_NAME)  
```
